EmailValidator/Validation/MailboxCheckValidation.php

```diff
@@ -0,0 +1,240 @@
+
+    /**
+     * MailboxCheckValidation constructor.
+     */
+    public function __construct()
+    {
+        if (!extension_loaded('intl')) {
+            throw new \LogicException(sprintf('The %s class requires the Intl extension.', __CLASS__));
+        }
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getError()
+    {
+        return $this->error;
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getWarnings()
+    {
+        return $this->warnings;
+    }
+
+    /**
+     * @return int
+     */
+    public function getLastResponseCode()
+    {
+        return $this->lastResponseCode;
+    }
+
+    /**
+     * @return int
+     */
+    public function getPort()
+    {
+        return $this->port;
+    }
+
+    /**
+     * @param int $port
+     */
+    public function setPort($port)
+    {
+        $this->port = $port;
+    }
+
+    /**
+     * @return int
+     */
+    public function getTimeout()
+    {
+        return $this->timeout;
+    }
+
+    /**
+     * @param int $timeout
+     */
+    public function setTimeout($timeout)
+    {
+        $this->timeout = $timeout;
+    }
+
+    /**
+     * @return string
+     */
+    public function getFromEmail()
+    {
+        return $this->fromEmail;
+    }
+
+    /**
+     * @param string $fromEmail
+     */
+    public function setFromEmail($fromEmail)
+    {
+        $this->fromEmail = $fromEmail;
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function isValid($email, EmailLexer $emailLexer)
+    {
+        $mxHosts = $this->getMXHosts($email);
+
+        $isValid = false;
+        foreach ($mxHosts as $mxHost) {
+            if ( ($isValid = $this->checkMailbox($mxHost, $this->port, $this->timeout, $this->fromEmail, $email)) ) {
+                break;
+            }
+        }
+
+        if ( ! $isValid ) {
+            $this->error = new IllegalMailbox($this->lastResponseCode);
+        }
+
+        return $this->error === null;
+    }
+
+    /**
+     * @param string $email
+     *
+     * @return array
+     */
+    protected function getMXHosts($email)
+    {
+        $variant = defined('INTL_IDNA_VARIANT_UTS46') ? INTL_IDNA_VARIANT_UTS46 : INTL_IDNA_VARIANT_2003;
+
+        $hostname = $email;
+        if ( false !== ($lastAtPos = strrpos($email, '@')) ) {
+            $hostname = substr($email, $lastAtPos + 1);
+        }
+        $hostname = rtrim(idn_to_ascii($hostname, IDNA_DEFAULT, $variant), '.') . '.';
+
+        $mxHosts = [];
+        $result = getmxrr($hostname, $mxHosts);
+        if ( ! $result ) {
+            $this->warnings[NoDNSMXRecord::CODE] = new NoDNSMXRecord();
+        }
+
+        return $mxHosts;
+    }
+
+    /**
+     * @param string $hostname
+     * @param int $port
+     * @param int $timeout
+     * @param string $fromEmail
+     * @param string $toEmail
+     * @return bool
+     */
+    protected function checkMailbox($hostname, $port, $timeout, $fromEmail, $toEmail)
```

Comment:
This params are already provided by properties of the class, except for `$hostname` and `$toEmail`.
Also, please make the method private.

Reply:
Maybe a better name, given the bool returned, would be `mailboxExist`.

Reply:
fix